### PR TITLE
Add korean support

### LINKED ISSUE #153: Korean support
Hello. I’m going to submit a PR for korean support, please review.

### PR DESCRIPTION
**Pull request by qbx2**
# Pull Request

## Related issue
Fixes #153 

## What does this PR do?
- Add korean support

## PR checklist
Please check if your PR fulfills the following requirements:
- [ ] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [ ] Have you read the contributing guidelines?
- [ ] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,9 @@ hebrew = []
 # allow japanese specialized tokenization
 japanese = ["dep:lindera"]
 
+# allow korean specialized tokenization
+korean = ["dep:lindera"]
```
Comment (ManyTheFish) [inline Cargo.toml:39]:
```suggestion
korean = ["dep:lindera", "lindera/ko-dic"]
```

File: Cargo.toml [hunk 2]
```diff
@@ -20,11 +20,11 @@ once_cell = "1.5.2"
 slice-group-by = "0.3.0"
 unicode-segmentation = "1.6.0"
 whatlang = "0.16.1"
-lindera = { version = "=0.16.0", features = ["ipadic"], optional = true }
+lindera = { version = "=0.17.0", features = ["ipadic", "ko-dic"], optional = true }
```
Comment (ManyTheFish) [inline Cargo.toml:23]:
```suggestion
lindera = { version = "=0.17.0", default-features = false, optional = true }
```

File: Cargo.toml [hunk 3]
```diff
@@ -35,6 +35,9 @@ hebrew = []
 # allow japanese specialized tokenization
 japanese = ["dep:lindera"]
```
Comment (ManyTheFish) [inline Cargo.toml:36]:
```suggestion
japanese = ["dep:lindera", "lindera/ipadic"]
```

**Review by ManyTheFish — CHANGES_REQUESTED** [review]
Hello @qbx2,
I requested some small changes to reduce the Library size when only Japanese is used or when only Korean is used.
I let you make the changes before re-reviewing your PR.
By the way, your PR is still in draft, don't you want to convert it?

**Comment by qbx2** [comment]
Actually, I’ve made this PR draft to add bench data.

**Comment by ManyTheFish** [comment]
@qbx2 yes you're right! We should add benchmark cases for Korean!
I let you do it, it's quite simple! Then we will be able to merge your PR.

**Review by ManyTheFish — APPROVED** [review]
Hey @qbx2,
this PR looks good to me, I let bors run the test and, if everything goes well, your PR will be merged automatically.

Thank you a lot for your spontaneous contribution! 😄 

bors merge

**Pull request merged**